lambda/app.py

```diff
@@ -58,104 +49,47 @@ def send_custom_metric( metric_name, dimensions, value, unit, namespace, timesta
-    run_stmt_response = glue_client.run_statement(
-        SessionId=session_id,
-        Code=f"df = spark.sql(\"{sql_stmt}\");df.show(df.count(),truncate=False)"
-    )
-    stmt_id = run_stmt_response["Id"]
-    logger.info(f"select files statement_id={stmt_id}")
-    stmt_response = wait_for_statement(session_id, run_stmt_response["Id"])
-    data_str = stmt_response["Statement"]["Output"]["Data"]["TextPlain"]
-    logger.info(stmt_response)
-    df = parse_spark_show_output(data_str)
-    df = df.applymap(int)
-    file_metrics = {
-        "avg_record_count": df.iloc[0]["avg_record_count"],
-        "max_record_count": df.iloc[0]["max_record_count"],
-        "min_record_count": df.iloc[0]["min_record_count"],
-        "avg_file_size": df.iloc[0]['avg_file_size'],
-        "max_file_size": df.iloc[0]['max_file_size'],
-        "min_file_size": df.iloc[0]['min_file_size'],
-    }
-    logger.info("file_metrics=")
-    logger.info(file_metrics)
-    # loop over file_metrics, use key as metric name and value as metric value
-    # loop over partition_metrics, use key as metric name and value as metric value
-    for metric_name, metric_value in file_metrics.items():     
-        logger.info(f"metric_name=files.{metric_name}, metric_value={metric_value.item()}")
+       
+def send_metrics(metrics: dict, namespace: str, table: Table, snapshot: Snapshot):
+    for metric_name, metric_value in metrics.items():
+        if not isinstance(metric_value, Number):
+            metric_value = metric_value.item()
+        logger.info(f"metric_name={namespace}.{metric_name}, metric_value={metric_value}")
         send_custom_metric(
-            metric_name=f"files.{metric_name}",
+            metric_name=f"{namespace}.{metric_name}",
             dimensions=[
-                {'Name': 'table_name', 'Value': f"{glue_db_name}.{glue_table_name}"}
+                {'Name': 'table_name', 'Value': f"{table.name()[1]}.{table.name()[2]}"}
             ],
-            value=metric_value.item(),
+            value=metric_value,
             unit='Bytes' if "size" in metric_name else "Count",
-            namespace=os.getenv('CW_NAMESPACE'),
+            namespace=cw_namespace,
             timestamp = snapshot.timestamp_ms,
         )
-    
 
-def send_partition_metrics(glue_db_name, glue_table_name, snapshot,session_id):
-    sql_stmt = f"select partition,record_count,file_count from glue_catalog.{glue_db_name}.{glue_table_name}.partitions"    
-    run_stmt_response = glue_client.run_statement(
-        SessionId=session_id,
-        Code=f"df = spark.sql(\"{sql_stmt}\");df.show(df.count(),truncate=False)"
-    )
+def send_files_metrics(table: Table, snapshot: Snapshot):
+    logger.info(f"send_files_metrics() -> snapshot_id={snapshot.snapshot_id}")
+    df = table.inspect.files().to_pandas()
```

Comment:
take into account that this method can be very expensive , for example a table with thousands of files this can be long , does this metric help with understanding anything ??

in the end if the table is partitioned what matters is the state of partitions and not avg  files across entire table

Reply:
Initially this metric was exposed. I guess it's a question for the actual template users whether to use it or not... 
@moryachok - wdyt ?

Reply:
Same could true for partitions as well - with thousands of partitions this could be expensive. We can do send_files_metrics and send_partitions_metrics optional

Reply:
i agree that their can be thousands of partitions but for this the best case would be 1 file per partition which would make it n files but im sure that would not be the case and in reality it much more than 1 file per partition.
their will always be more files than partitions